Mathlib/CategoryTheory/Abelian/GrothendieckCategory/ModuleEmbedding/GabrielPopescu.lean

```diff
@@ -0,0 +1,150 @@
+/-
+Copyright (c) 2025 Markus Himmel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Markus Himmel
+-/
+import Mathlib.Algebra.Category.ModuleCat.Injective
+import Mathlib.CategoryTheory.Abelian.GrothendieckAxioms.Connected
+import Mathlib.CategoryTheory.Abelian.GrothendieckCategory.Coseparator
+import Mathlib.CategoryTheory.Preadditive.Injective.Preserves
+import Mathlib.CategoryTheory.Preadditive.LiftToFinset
+import Mathlib.CategoryTheory.Preadditive.Yoneda.Limits
+
+/-!
+# The Gabriel-Popescu theorem
+
+We prove the following Gabriel-Popescu theorem: if `C` is a Grothendieck abelian category and
+`G` is a separator, then the functor `preadditiveCoyonedaObj G : C ⥤ ModuleCat (End R)ᵐᵒᵖ` sending
+`X` to `Hom(G, X)` is fully faithful and has an exact left adjoint.
+
+We closely follow the elementary proof given by Barry Mitchell.
+
+## Future work
+
+The left adjoint `tensorObj G` actually exists as soon as `C` is cocomplete and additive, so the
+construction could be generalized.
+
+The theorem as stated here implies that `C` is a Serre quotient of `ModuleCat (End R)ᵐᵒᵖ`.
```

Comment:
With the lemma `isLocalization_serreW_kernel_of_leftAdjoint` from #22741, it should be possible to obtain that `tensorObj G` is a localization functor.